### ISSUE TITLE
Drop usage of Horizon

### ISSUE THREAD

**Issue by tyvdh** [astroband/astrograph#158]
The current app as far as I can see makes heavy usage of the official Horizon API. You've written a lot if not all of the resolvers for converting from the core database XDRs to a usable data format, all that's left is to find an alternative database to the one Stellar uses in their Horizon endpoint to store the deserialized XDRs. I assume that's what dgraph is trying to do? Haven't poked around too much but I definitely submit that elasticsearch will likely be the best for this. The hardest bit will be the cost. Right now the Horizon db is nearly 1TB of data so we can only assume an Elasticsearch alternative will be similar and just continue to grow. 🤑 

Not sure what the solution to all of this is but as I have explored this space quite a lot myself I'd welcome a discussion so just opening a ticket to hear your thoughts and plans for transitioning away from using Horizon in what you call a Horizon replacement.

**Comment by gzigzigzeo** [comment]
Hi!

Yes, you are correct, DGraph (at least, for now) can not handle required amount of data, we need another storage. Elasticsearch looks as the best fit. We already have written ingestion service (WIP, in golang, will OSS it soon). It works (ledgers, transactions, operations), but we have not yet implemented some secondary entities (like operation results). 

We are concerned about possible infra costs, and we intend to measure it by ingesting good piece of ledger history and see what real database size will be, how fast will it work and how much resources will it require to work reasonably fast. For now, it looks like ES indicies disk size is even little smaller than Postgresql database.

Frankly, I am not sure that all users are going to need full history all the time, so, we could implement retention period to keep cost low, as core does. And those who want full history might use their own setup, that's always an option.

We also considering the possibility to use various storages at once: like, ES for historical queries, and some kind graph database, containing significantly reduced data set for graph related analysis.

Besides graph storages, we considered Google BigTable/BigQuery and Timescaledb. We will be happy to hear if you have any thoughts about alternative storage engines, probably you have experimented with any others?

**Comment by tyvdh** [comment]
Exciting! Sounds like you put a lot of thought and work into it.

The only ones I've worked with are Elasticsearch and IPFS, ultimately landing on Elasticsearch. IPFS is cool as a ledger combing tool but useless for actually searching things historically or with any level of complexity. We use a combination of Elasticsearch and Memsql @Baremetrics which compiles and delivers many TB of data without a hitch. It is wicked expensive though. Ideally yeah like you said maybe free access to portions of the db or all of the testnet with either self-hosted or paid access to the full public network. I've talked to quite a few developers and they were all open to the idea of a paid version of my graphql service as long as it was fast and production ready. I'm personally in the same boat. Would love to see a version of this make it to a paid maturity level. It costs so much already to setup and run your own core and horizon instances that setting up a method of access into the database without having to go through that headache or cost would easily be worth ~$15/mo.
